tests/consensus_tests/auth_tests/test_jwt_validation.py

```diff
@@ -158,3 +168,92 @@ def test_payload_filters_queries():
     res_expected.raise_for_status()
 
     assert res.json()["result"] == res_expected.json()["result"]
+
+
+@pytest.mark.parametrize(
+    "prefetch",
+    [
+        None,
+        {"query": [0.1, 0.2, 0.3, 0.4]},
+        [{"query": [0.1, 0.2, 0.3, 0.4]}, {"limit": 100}],
+    ],
+)
+def test_payload_filters_query(berlin_token, prefetch):
+    res = requests.post(
+        f"{REST_URI}/collections/{COLL_NAME}/points/query",
+        json={
+            "prefetch": prefetch,
+            "query": [0.1, 0.2, 0.3, 0.4],
+            "limit": 100,
+            "with_payload": True,
+        },
+        headers={"Authorization": f"Bearer {berlin_token}"},
+    )
+
+    cities_in_payload = set(
+        point["payload"].get("city") for point in res.json()["result"]
+    )
+
+    assert cities_in_payload == {"Berlin"}
```

Comment:
Are we asserting somewhere else that points exist with different cities than Berlin in the test dataset?

Reply:
It's in the collection fixture, but I'll add a quick call to assert it here